boxsdk/object/upload_session.py

```diff
@@ -0,0 +1,165 @@
+# coding: utf-8
+from __future__ import unicode_literals, absolute_import
+
+import base64
+import hashlib
+import json
+
+from .base_object import BaseObject
+from ..pagination.limit_offset_based_dict_collection import LimitOffsetBasedDictCollection
+
+
+class UploadSession(BaseObject):
+    _item_type = 'upload_session'
+    _parent_item_type = 'file'
+
+    def get_url(self, *args):
+        """
+        Base class override. Endpoint is a little different - it's /files/upload_sessions.
+
+        :rtype:
+            `unicode`
+        """
+        return self._session.get_url(
+            '{0}s/{1}s'.format(self._parent_item_type, self._item_type),
+            self._object_id,
+            *args
+        ).replace(self.session.api_config.BASE_API_URL, self.session.api_config.UPLOAD_URL)
+
+    def get_parts(self, limit=None, offset=None):
+        """
+        Get a list of parts uploaded so far.
+
+        :param limit:
+            The maximum number of items to return per page. If not specified, then will use the server-side default.
+        :type limit:
+            `int` or None
+        :param offset:
+            The index at which to start returning items.
+        :type offset:
+            `int` or None
+        :param fields:
+            Fields to include on the returned items.
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            Returns a `list` of parts uploaded so far.
+        :rtype:
+            `list` of `dict`
+        """
+        return LimitOffsetBasedDictCollection(
+            session=self.session,
+            url=self.get_url('parts'),
+            limit=limit,
+            offset=offset,
+            fields=None,
+            return_full_pages=False,
+        )
+
+    def upload_part_bytes(self, part_bytes, offset, total_size, part_content_sha1=None):
+        """
+        Upload a part of a file.
+
+        :param part_bytes:
+            Part bytes
+        :type part_bytes:
+            `bytes`
+        :param offset:
+            Offset, in number of bytes, of the part compared to the beginning of the file. This number should be a
+            multiple of the part size.
+        :type offset:
+            `int`
+        :param total_size:
+            The size of the file that this part belongs to.
+        :type total_size:
+            `int`
+        :param part_content_sha1:
+            SHA-1 hash of the part's content. If not specified, this will be calculated.
+        :type part_content_sha1:
+            `unicode` or None
```

Comment:
Is this the right type?  Below you encode the hash in a way that I think requires `bytes`